.ci/magician/cmd/request_service_reviewers.go

```diff
@@ -0,0 +1,149 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package cmd
+
+import (
+	"fmt"
+	"magician/github"
+	"math/rand"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/magic-modules/tools/issue-labeler/labeler"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+// requestServiceReviewersCmd represents the requestServiceReviewers command
+var requestServiceReviewersCmd = &cobra.Command{
+	Use:   "request-service-reviewers PR_NUMBER",
+	Short: "Assigns reviewers based on the PR's service labels.",
+	Long: `This command requests (or re-requests) review based on the PR's service labels.
+
+	If a PR has more than 3 service labels, the command will not do anything.
+	`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		prNumber := args[0]
+		fmt.Println("PR Number: ", prNumber)
+
+		gh := github.NewClient()
+		execRequestServiceReviewers(prNumber, gh, labeler.EnrolledTeamsYaml)
+	},
+}
+
+func execRequestServiceReviewers(prNumber string, gh GithubClient, enrolledTeamsYaml []byte) {
+	pullRequest, err := gh.GetPullRequest(prNumber)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	enrolledTeams := make(map[string]labeler.LabelData)
+	if err := yaml.Unmarshal(enrolledTeamsYaml, &enrolledTeams); err != nil {
+		fmt.Printf("Error unmarshalling enrolled teams yaml: %s", err)
+		os.Exit(1)
+	}
+
+	requestedReviewers, err := gh.GetPullRequestRequestedReviewers(prNumber)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	previousReviewers, err := gh.GetPullRequestPreviousReviewers(prNumber)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// If more than three service labels are impacted, don't request reviews.
+	// Only request reviews from unique service teams.
+	githubTeams := make(map[string]struct{})
+	teamCount := 0
+	for _, label := range pullRequest.Labels {
+		if !strings.HasPrefix(label.Name, "service/") || label.Name == "service/terraform" {
+			continue
+		}
+		teamCount += 1
+		if labelData, ok := enrolledTeams[label.Name]; ok {
+			githubTeams[labelData.Team] = struct{}{}
+		}
+	}
+
+	if teamCount > 3 {
+		fmt.Println("Provider-wide change (>3 services impacted); not requesting service team reviews")
+		return
+	}
+
+	// For each service team, check if one of the team members is already a reviewer. Rerequest
+	// review if there is and choose a random reviewer from the list if there isn't.
+	toRequest := []string{}
+	requestedReviewersMap := make(map[string]struct{})
+	for _, reviewer := range requestedReviewers {
+		requestedReviewersMap[reviewer.Login] = struct{}{}
+	}
+
+	previousReviewersMap := make(map[string]struct{})
+	for _, reviewer := range previousReviewers {
+		previousReviewersMap[reviewer.Login] = struct{}{}
```

Comment:
```suggestion
		previousReviewersMap[reviewer.Login] = map[string][]User
```
Make object declaration more specific

Reply:
`map[string]struct{}` is the standard way of doing a set in Go - it's slightly more efficient than `map[string]bool`. There isn't a user list here.

Reply:
Apologies for the confusion.

My concern here is that we are initializing `previousReviewersMap[reviewer.Login]` to `struct{}{}`. For readability I would prefer to initialize it to the actually type we intend to use. That way when a new contributor is reading they don't have to gain context on the type definitions and it takes up the same space.

I won't block on this.

Reply:
previousReviewersMap is a `map[string]struct{}` - this line is setting the key `reviewer.Login` (a string) to `struct{}{}` (an empty object of type `struct`), which is the standard way of initializing a set in Go - see https://www.reddit.com/r/golang/comments/xfc33s/when_will_go_get_sets/ for some discussion.

`struct{}{}` is the intended value here; it's never used, it's just a placeholder to indicate that the reviewer is in the set.

I wouldn't want to change this to a `map[string][]User` because we're not using the value, so it doesn't need to hold meaningful data.

I'll rename these variables to `Set` instead of `Map` for clarity.

Please let me know if I'm missing / misunderstanding anything.

Reply:
Ah, you're totally right here. My misunderstanding. I was conflating this with something else in my head. 

Thanks for the clarifications!